Mathlib/RingTheory/MvPowerSeries/Basic.lean

```diff
@@ -719,6 +720,76 @@ theorem coeff_prod [DecidableEq σ]
       simp only [add_right_inj] at huv
       exact h rfl huv.symm
 
+/-- The `d`th coefficient of a power of a multivariate power series
+is the sum, indexed by `finsuppAntidiag (Finset.range n) d`, of products of coefficients  -/
+theorem coeff_pow [DecidableEq σ] (f : MvPowerSeries σ R) {n : ℕ} (d : σ →₀ ℕ) :
+    coeff R d (f ^ n) =
+      ∑ l in finsuppAntidiag (Finset.range n) d,
+        ∏ i in Finset.range n, coeff R (l i) f := by
+  suffices f ^ n = (Finset.range n).prod fun _ ↦ f by
+    rw [this, coeff_prod]
+  rw [Finset.prod_const, card_range]
+
+/-- The degree of a monomial. -/
+def degree (d : σ →₀ ℕ) : ℕ := d.sum fun _ ↦ id
+
+@[simp]
+theorem degree_zero : degree (0 : σ →₀ ℕ) = 0 := by
+  simp only [degree, sum_zero_index]
+
+theorem degree_add (d d' : σ →₀ ℕ) :
+    degree (d + d') = degree d + degree d' := by
+  classical
+  simp only [degree, mem_union, mem_support_iff, ne_eq, id_eq, implies_true, sum_add_index]
+
+theorem degree_eq_zero_iff (d : σ →₀ ℕ) : degree d = 0 ↔ d = 0 := by
```

Comment:
Indeed `exact MvPolynomial.degree_eq_zero_iff _` proves this (after importing it). The really good thing to do would be to extract all this results to a new file, but if everything you need already exists in the `MvPolynomial` you can just use it.

Reply:
See above.